File: frontend/src/types/ioc.ts
export type IOCType = 'email' | 'domain' | 'ip' | 'url' | 'hash' | 'file' | 'username'
export type IOCStatus = 'active' | 'blocked' | 'remediated' | 'fp'
export type TLPLevel = 'white' | 'green' | 'amber' | 'red'

export interface IOC {
  id: string
  incident_id: string
  ioc_type: IOCType
  value: string
  confidence: number
  status: IOCStatus
  enrichment: Record<string, unknown>
  tlp_level: TLPLevel
  tags: string[]
  created_at: string
  updated_at: string
}

export interface CreateIOCPayload {
  ioc_type: IOCType
  value: string
  confidence?: number
  tlp_level?: TLPLevel
  tags?: string[]
}

export interface UpdateIOCPayload {
  status?: IOCStatus
  confidence?: number
  tlp_level?: TLPLevel
  tags?: string[]
}

export interface DetectedIOC {
  ioc_type: IOCType
  value: string
}

export const IOC_TYPE_ICONS: Record<IOCType, string> = {
  email:    'e-mail_color.svg',
  domain:   'globe_with_meridians_color.svg',
  ip:       'electric_plug_color.svg',
  url:      'link_color.svg',
  hash:     '#',
  file:     'page_facing_up_color.svg',
  username: 'bust_in_silhouette_color.svg',
}

export const IOC_STATUS_COLORS: Record<IOCStatus, string> = {
  active:    'chip-red',
  blocked:   'chip-yellow',
  remediated:'chip-green',
  fp:        'chip-muted',
}

export const TLP_COLORS: Record<TLPLevel, string> = {
  white: 'chip-muted',
  green: 'chip-green',
  amber: 'chip-yellow',
  red:   'chip-red',
}

File: backend/app/schemas/ioc.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel


class IOCCreate(BaseModel):
    ioc_type: Literal["email", "domain", "ip", "url", "hash", "file", "username"]
    value: str
    description: str | None = None
    confidence: int = 50
    status: Literal["active", "blocked", "remediated", "fp"] = "active"
    tlp_level: Literal["red", "amber", "green", "white"] = "red"
    tags: list[str] = []


class IOCUpdate(BaseModel):
    description: str | None = None
    confidence: int | None = None
    status: Literal["active", "blocked", "remediated", "fp"] | None = None
    tlp_level: Literal["red", "amber", "green", "white"] | None = None
    tags: list[str] | None = None


class IOCBulkImport(BaseModel):
    """Paste raw text — server auto-detects IOC types."""
    text: str


class IOCOut(BaseModel):
    id: UUID
    incident_id: UUID
    ioc_type: str
    value: str
    description: str | None
    confidence: int
    status: str
    first_seen: datetime
    added_by: UUID | None
    enrichment: dict
    tlp_level: str
    tags: list[str]
    created_at: datetime

    model_config = {"from_attributes": True}


class IOCDetected(BaseModel):
    """Auto-detected IOC suggestion — not yet saved."""
    ioc_type: str
    value: str
